lib/spack/spack/test/concretize_requirements.py

```diff
@@ -391,6 +391,26 @@ def test_preference_adds_new_version(
     assert not s3.satisfies("@2.3")
 
 
+def test_external_adds_new_version_that_is_preferred(concretize_scope, test_repo):
```

Comment:
Can we add a description of the tests similar to:
```suggestion
def test_external_adds_new_version_that_is_preferred(concretize_scope, test_repo):
    """Tests that we can use a version, not declared in package recipes, as the preferred version if that
    version appears in an external spec.
    """
```
?

Reply:
(I have added the docstring, although this comment is not marked as outdated)